archethic/contracts/factory.exs

```diff
@@ -194,13 +182,29 @@ export fun get_chargeable_htlc(end_time, user_address, pool_address, secret_hash
       end
       \\\"""
     end
-    
-    fun query_evm_apis(endpoints, method, headers, body) do
+
+    fun query_evm_apis(endpoints, body) do
       requests = []
       for endpoint in endpoints do
-        requests = List.append(requests, url: endpoint, method: method, headers: headers, body: body)
+        requests = List.append(requests, url: endpoint, method: "POST", headers: ["Content-Type": "application/json"], body: Json.to_string(body))
+      end
+
+      responses = Http.request_many(requests, false)
+
+      valid_http_request? = false
+      valid_response = nil
+      for res in responses do
+        if !valid_http_request? && res.status == 200 && Json.is_valid?(res.body) do
+          valid_response = Json.parse(res.body)
+          valid_http_request? = true
+        end
       end
-      Http.request_many(requests, false)
+
+      if !valid_http_request? do
+        throw message: "Cannot fetch EVM RPC endpoints", code: 500
```

Comment:
It could be interesting to add `res.status` as data of the error, this would help determining why the request failed (bad request, timeout, fetch size exceeded ...)
Same suggestion for uco_pool.exs and token_pool.exs